### PR TITLE
Update node to version 16

### PR DESCRIPTION
**Pull request by johnsonm325**
## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [x] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by bastilian — APPROVED** [review]
LGTM!

File: package.json [hunk 1]
```diff
@@ -49,45 +49,46 @@
...
-    "eslint-config-prettier": "^7.1.0",
-    "eslint-plugin-react": "^7.25.3",
+    "eslint": "^8.8.0",
+    "eslint-config-prettier": "^8.3.0",
+    "eslint-plugin-react": "^7.28.0",
     "identity-obj-proxy": "^3.0.0",
-    "jest": "^26.6.3",
-    "moment": "^2.27.0",
+    "jest": "^27.5.1",
+    "moment": "^2.29.1",
```
Comment (Hyperkid123) [inline package.json:76]: This should be moved to dependencies

File: package.json [hunk 2]
```diff
@@ -49,45 +49,46 @@
...
     "redux-mock-store": "^1.5.4",
-    "redux-promise-middleware": "^5.1.1",
-    "regenerator-runtime": "^0.12.1",
-    "stylelint": "^13.13.1",
-    "stylelint-config-recommended-scss": "^4.2.0",
-    "stylelint-scss": "^3.21.0",
-    "webpack": "^5.53.0",
-    "webpack-bundle-analyzer": "^4.4.2",
-    "webpack-cli": "^4.8.0"
+    "redux-promise-middleware": "^6.1.2",
```
Comment (Hyperkid123) [inline package.json:80]: ^^

**Review by Hyperkid123 — COMMENTED** [review]
You have installed the dependencies using node 14 but require node 16 in CI. This may cause issues. Can reinstall the dependencies using nod16? You can check the lockfile. It should have version 2.

**Comment by Hyperkid123** [comment]
@johnsonm325 I think that there might be some async tasks failing in the tests. I see some async redux actions in a few of the related components. Async tests without proper mocks can teardown the whole testing environment.

**Comment by johnsonm325** [comment]
@bastilian @Hyperkid123 This is ready for a proper review.

**Review by Hyperkid123 — APPROVED** [review]

**Pull request merged**